components/toast/script.js

```diff
@@ -155,18 +236,30 @@ class GamefaceToast extends BaseComponent {
     handleTimeOut() {
         if (this.hideTimeOut) clearTimeout(this.hideTimeOut);
 
-        if (this.timeout > 0) {
-            this.hideTimeOut = setTimeout(this.hide, this.timeout);
+        if (this.state.timeout > 0) {
+            this.hideTimeOut = setTimeout(this.hide, this.state.timeout);
         }
     }
 
     /**
      * Setups the close button of the toast
      */
-    handleCloseButton() {
-        const closeButton = this.querySelector('.guic-toast-close-btn');
-        if (closeButton.firstElementChild.clientWidth && closeButton.firstElementChild.clientHeight) {
-            closeButton.addEventListener('click', this.hide);
+    async handleCloseButton() {
+        await components.waitForFrames(() => {
+            if (this._closeButton.firstElementChild.clientWidth && this._closeButton.firstElementChild.clientHeight) {
+                this._closeButton.addEventListener('click', this.hide);
+            }
+        }, 2);
```

Comment:
```suggestion
        await components.waitForFrames(() => {
            const { clientWidth, clientHeight } = this._closeButton.firstElementChild;
            if (clientWidth && clientHeight) this._closeButton.addEventListener('click', this.hide);
        }, 2);
```

Reply:
Why do you need to wait for 2 frames and not 3 or 1?

Reply:
I think there was some error in the player where it couldn't read the clientWidth and Height so Misho suggested that I wrap it inside the `waitForFrames`  function and It just worked with 2 frames .